File: src/main/resources/generators/scala/native/ListWrapper.scala
import java.util.Date

class ListWrapper[A](private val inner: List[A]) {
  def nonEmpty(): Boolean = inner.nonEmpty

  def isEmpty(): Boolean = inner.isEmpty

  def head: A = inner.head

  def forall(f: A => Boolean): Boolean = inner.forall(f)

  def exists(f: A => Boolean): Boolean = inner.exists(f)

  def foldLeft[B](startValue: B, f: (B, A) => B): B = inner.foldLeft(startValue)(f)
}

File: src/main/resources/generators/ts/native/ListWrapper.ts
export class ListWrapper<A> {
    private inner: Array<A>;

    constructor(inner: Array<A>) {
        this.inner = inner;
    }

    nonEmpty(): boolean {
        return this.inner && this.inner.length > 0;
    }

    isEmpty(): boolean {
        return !this.nonEmpty();
    }

    get head(): A {
        return this.inner ? this.inner[0] : null;
    }

    randomElement(): A {
        // Not really random, it is for the example currently.
        return this.inner ? this.inner[0] : null;
    }

    forall(f: (A) => Boolean): Boolean {
        return this.inner.every(f);
    }

    exists(f: (A) => Boolean): Boolean {
        return this.inner.some(f);
    }

    foldLeft<B>(startValue: B, f: (B, A) => B): B {
        let acc = startValue;
        for (let i = 0 ; i < this.inner.length ; i++) {
            acc = f(acc, this.inner[i]);
        }
        return acc;
    }
}